fabric-data-generation-api-v1/src/main/java/net/fabricmc/fabric/api/datagen/v1/DataGeneratorEntrypoint.java

```diff
@@ -31,4 +35,15 @@ public interface DataGeneratorEntrypoint {
 	 * @param fabricDataGenerator The {@link FabricDataGenerator} instance
 	 */
 	void onInitializeDataGenerator(FabricDataGenerator fabricDataGenerator);
+
+	/**
+	 * Return the {@link ModContainer} of the mod the data is being generated for.
+	 * A null return will run the data generation using the {@link ModContainer} that registered the current entrypoint.
+	 *
+	 * @return a {@link ModContainer} or null
+	 */
+	@Nullable
+	default ModContainer getEffectiveModContainer() {
```

Comment:
On second thoughts it may be better to just return a nullable string of the modid, and find the mod container in the impl.